spec/Candid.md

```diff
@@ -70,7 +70,7 @@ This is a summary of the grammar proposed:
 <actortype> ::= { <methtype>;* }
 <methtype>  ::= <name> : (<functype> | <id>)
 <functype>  ::= <tuptype> -> <tuptype> <funcann>*
-<funcann>   ::= oneway | query
+<funcann>   ::= oneway | query | composite_query
```

Comment:
I still think this is a very badly chosen name. My suggestion (that came too late to be considered in the replica naming contest) was _report_.

Reply:
I wish it's named `query_tmp_hack`

Reply:
QUERY. As in, a larger query.

Reply:
QUEyr. It's bigger than `query`, but not big enough (only the same subnet). The spelling is wrong, because it has strange await semantics...

Reply:
@ggreif: I associate "report" with documents and graphs, not queries. If I was a developer, I would be confused by that name. I think the name has to contain the word "query" or at the very least "queyr" 😄 

@chenyan-dfinity: "because it has strange await semantics..."
There is a trade-off between strange await semantics and strange query semantics. Community voted to keep the proper query semantics sacrificing a corner case of self-call await. We also surveyed the use cases of people interested in composite queries and all of the uses cases involved calling different canisters, where await works as expected.

Reply:
Here are my further naming suggestions
- `inquisition` (religious undertone)
- `interrogation` (too long, menacing)
- `quiz` (many questions, sounds fitting)

Reply:
We could just ask Dom. That's how naming debates have been resolved in the past :-)

Reply:
I still quite like `enquiry` or `inquiry`, but never mind.

Reply:
I hope this discussion is not blocking the PR.